runelite-client/src/main/java/net/runelite/client/plugins/puzzlesolver/solver/PuzzleState.java

```diff
@@ -53,26 +54,32 @@ public PuzzleState(int[] pieces)
 			throw new IllegalStateException("Piece array does not have the right dimensions");
 		}
 
-		for (int i = 0; i < pieces.length; i++)
-		{
-			if (pieces[i] == BLANK_TILE_VALUE)
-			{
-				emptyPiece = i;
-			}
-		}
+		this.pieces = pieces;
+
+		this.emptyPiece = findEmptyPiece();
 
 		if (emptyPiece == -1)
 		{
 			throw new IllegalStateException("Incorrect empty piece passed in!");
 		}
-
-		this.pieces = pieces;
 	}
 
-	private PuzzleState(PuzzleState state)
+	public PuzzleState(PuzzleState state)
 	{
 		this.pieces = Arrays.copyOf(state.pieces, state.pieces.length);
-		this.emptyPiece = state.emptyPiece;
+		this.emptyPiece = state.findEmptyPiece();
+	}
+
+	private int findEmptyPiece()
```

Comment:
I would make this method not return anything and instead set the ``this.emptyPiece`` inside.
And then ``throw new IllegalStateException("Incorrect empty piece passed in!");`` after the loop.
Then remove the exception throwing for emptyPiece in the constructor.